includes/class-wc-stripe-intent-controller.php

```
@@ -723,6 +723,19 @@ public function update_order_status_ajax() {
 				],
 				200
 			);
+		} catch ( WC_Stripe_Payment_Cancelled_Exception $e ) {
+			// Customer-initiated cancellation (e.g. closed Klarna popup). Do not fail the
+			// order — leave it retryable and return an error so the frontend can notify the customer.
+			WC_Stripe_Logger::info(
+				'Payment cancelled by customer via AJAX for order: ' . $order_id . '. Reason: ' . $e->getMessage()
+			);
```

PHPMD (2.15.0)

[error] Avoid using static access to class 'WC_Stripe_Logger' in method 'update_order_status_ajax'. (undefined)

(StaticAccess)